src/fauxtoshop.cpp

```diff
@@ -1,517 +1,487 @@
+                        // If the edge is larger than the edgeThreshold, this is an edge
+                        if (edge > edgeThreshold){
+                            isEdge = true;
+                            blackWhiteGrid[i][j] = BLACK;
+                        } else {
+                            blackWhiteGrid[i][j] = WHITE;
+                        }
+                    }
+                }
+            }
+        }
+    }
+    img.fromGrid(blackWhiteGrid);
+}
+
+/*
+ * Function: applyGreenScreenFilter
+ * Usage: Paste a "sticker" image on top of a "background" image, but ignore any part of the sticker
+ * that is close to pure green in color
+ * Function:
+ * Image in the main menu is the background image.
+ * a. Prompt user to specify a new image file name to be the sticker image. Re-prompt if open fails
+ * b. Prompt user to specify tolerance for green (integer between 0 and 100 (inclusive) that isn't
+ * pure green (0x00FF00)
+ * c. Prompt user to specify location as (row,col) for where to place the sticker image. Reprompt
+ * if the input is invalid. If the user hits enter, they can specify the location with a mouse click
+ * d. Place the image in the location specified. Loop over each pixel, any pixel on the sticker
+ * image with difference greater than threshold from pure green will be copied onto the background,
+ * otherwise use the background image
+ * Sticker imagecan be cut off on the bottom or right edge(s) if it cannot completely fit on the
+ * background.
+ * Params: original - the background GBufferedImage, sticker - GBufferedImage, gw - GWindow,
+ * row - where the sticker is place, col - where the sticker is place
+ * ------------------------------------------------------------------------------------------------
+ * Returns: nothing. Void function
+*/
+void applyGreenScreenFilter(GBufferedImage& img){
+    // Prompt user to specify a new image file name to be the sticker image
+    GBufferedImage sticker;
+    nameImageFileToOpen(
+                sticker, "Enter name of image file to open ('?' to open file selector): ", false);
+    // Prompt user to specify tolerance for green that isn't pure green
+    int greenTolerance = getValidInteger(
+                "Enter the green tolerance as a between 0 and 100 (inclusive): ", 0, 100);
+    // Prompt user to specify location as a vector for where to place the sticker image
+    int row;
+    int col;
+    validateLocationInput(row, col);
+    Grid<int> original = img.toGrid();
+    // Create a new Grid with the same dimensions as the sticker
+    Grid<int> stickerImage = sticker.toGrid();
+    // Iterate through each pixel
+    for (int i = 0; i < stickerImage.numRows(); i++){
+        for (int j = 0; j < stickerImage.numCols(); j++) {
+            // Identify pixels that are not pure green for green screen effect
+            int nonGreenScreen = calculateRBGColourDifference(stickerImage[i][j], GREEN);
```

Comment:
Maybe rename `nonGreenScreen` to something like `colourDifference`? That would describe its purpose better